crates/ruff_python_ast/src/newlines.rs

```diff
@@ -0,0 +1,192 @@
+use std::iter::FusedIterator;
+
+/// Extension trait for [`str`] that provides a [`UniversalNewlineIterator`].
+pub trait StrExt {
+    fn universal_newlines(&self) -> UniversalNewlineIterator<'_>;
+}
```

Comment:
I went ahead and added an extension trait, but if you have any feedback @MichaReiser happy to address in a follow-up.